File: Baekjoon/Swift/BOJ_PS/BOJ_PS/Stack-Queue/Q10845.swift
//
//  main.swift
//  BOJ_PS
//
//  Created by 김태훈 on 2021/05/06.
//  큐 / 큐

let n = Int(readLine()!)!
var q = [Int]()

for _ in 0..<n {
    let input = readLine()!.split(separator: " ").map { String($0) }
    let cmd = input[0]
    
    if cmd == "push" {
        q.append(Int(input[1])!)
    } else if cmd == "pop" {
        print(q.isEmpty ? -1 : q.removeFirst())
    } else if cmd == "size" {
        print(q.count)
    } else if cmd == "empty" {
        print(q.isEmpty ? 1 : 0)
    } else if cmd == "front" {
        print(q.isEmpty ? -1 : q.first!)
    } else if cmd == "back" {
        print(q.isEmpty ? -1 : q.last!)
    }
}

File: Baekjoon/Python/BOJ_PS/Stack-Queue/Q10845.py
# 큐 / 큐 
import sys

input = sys.stdin.readline 
n = int(input()) 
q = []
for _ in range(n):
    cmd = input().split()
    if cmd[0] == 'push':
        q.append(int(cmd[1]))
    elif cmd[0] == 'pop':
        if q: print(q.pop(0))
        else: print(-1)
    elif cmd[0] == 'size':
        print(len(q))
    elif cmd[0] == 'empty':
        if q: print(0)
        else: print(1)
    elif cmd[0] == 'front':
        if q: print(q[0])
        else: print(-1)
    elif cmd[0] == 'back':
        if q: print(q[len(q)-1])
        else: print(-1)
